src/components/categories-page/index.tsx

```diff
@@ -0,0 +1,451 @@
+                { name: 'Confined Space Systems' }
+            ],
+            title: 'Rescue Subcategories'
+        },
+        {
+            category: 'First Aid',
+            subCategory: [
+                { name: 'Anchorage' },
+                { name: 'Bosun Chairs' },
+                { name: 'Carabiners & Hardware' },
+                { name: 'Confined Space Systems' }
+            ],
+            title: 'First Aid Subcategories'
+        },
+        {
+            category: 'Tools',
+            subCategory: [
+                { name: 'Anchorage' },
+                { name: 'Bosun Chairs' },
+                { name: 'Carabiners & Hardware' },
+                { name: 'Confined Space Systems' }
+            ],
+            title: 'Tools Subcategories'
+        },
+        {
+            category: 'Bags, Buckets, & Storage',
+            subCategory: [
+                { name: 'Anchorage' },
+                { name: 'Bosun Chairs' },
+                { name: 'Carabiners & Hardware' },
+                { name: 'Confined Space Systems' }
+            ],
+            title: 'Bags, Buckets, & Storage Subcategories'
+        },
+        {
+            category: 'Devices & Testing',
+            subCategory: [
+                { name: 'Anchorage' },
+                { name: 'Bosun Chairs' },
+                { name: 'Carabiners & Hardware' },
+                { name: 'Confined Space Systems' }
+            ],
+            title: 'Devices & Testing Subcategories'
+        }
+    ];
+
+    const url = '/fall-protection';
+
+    const infiniteImages = useMemo(() => {
+        if (images.length < 2) return images;
+
+        return [images[images.length - 1], ...images, images[0]];
+    }, [images]);
+
+    const [currentIndex, setCurrentIndex] = useState(1);
+    const [isPaused, setIsPaused] = useState(false);
+    const [isTransitionEnabled, setIsTransitionEnabled] = useState(true);
+
+    const totalSlides = infiniteImages.length;
+
+    const handleTransitionEnd = useCallback(() => {
+        if (currentIndex === 0) {
+            setIsTransitionEnabled(false);
+            setCurrentIndex(totalSlides - 2);
+        } else if (currentIndex === totalSlides - 1) {
+            setIsTransitionEnabled(false);
+            setCurrentIndex(1);
+        }
+    }, [currentIndex, totalSlides]);
+
+    useEffect(() => {
+        if (!isTransitionEnabled) {
+            const timer = setTimeout(() => {
+                setIsTransitionEnabled(true);
+            }, 10);
+
+            return () => clearTimeout(timer);
+        }
+    }, [isTransitionEnabled]);
+
+    const nextSlide = useCallback(() => {
+        if (isTransitionEnabled) {
+            setCurrentIndex((prev: any) => (prev + 1) % totalSlides);
+        }
+    }, [totalSlides, isTransitionEnabled]);
+
+    const prevSlide = useCallback(() => {
+        if (isTransitionEnabled) {
+            setCurrentIndex((prev: any) => (prev - 1 + totalSlides) % totalSlides);
```

Comment:
Do not use any.
Please define the type.
```suggestion
            setCurrentIndex((prev: number) => (prev - 1 + totalSlides) % totalSlides);
```